specification/security/_types/RoleMappingRule.ts

```diff
@@ -33,14 +34,11 @@ export class RoleMappingRule {
 /**
  * @variants container
  */
-export class FieldRule {
-  username?: Name
+export class FieldRule
+  implements AdditionalProperties<string, UserDefinedValue>
+{
+  username?: Names
```

Comment:
According to the [source](https://github.com/elastic/elasticsearch/blob/fe1083f6c5f29e1fc28b82c1b71be9b7c1ec7a73/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/UserRoleMapper.java#L55) this should be a string, I'd go for `Name` instead.

Reply:
Multiple usernames work and are documented, see `"username": [ "es-admin", "es-system" ]` in https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-put-role-mapping.html

Reply:
My bad, the API does allow for a list!